pkg/actuators/client/client.go

```diff
@@ -20,86 +20,407 @@ import (
+	if err != nil {
+		return "", err
+	}
+
+	// Check if resourceGroupList is not nil, in case of a 502, etc
+	if resourceGroupList != nil {
+		var resourceGroupID string
+		for _, eachResource := range resourceGroupList.Resources {
+			if *eachResource.Name == resourceGroupName {
+				// Get Resource Group ID
+				resourceGroupID = *eachResource.ID
+				return resourceGroupID, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("Could not retrieve Resource Group ID of name: %v", resourceGroupName)
+}
+
+// GetSubnetIDbyName retrives a Subnet ID
+func (c *ibmCloudClient) GetSubnetIDbyName(subnetName string, resourceGroupID string) (string, error) {
+	// Initialize List Subnets Options
+	subnetOption := c.vpcService.NewListSubnetsOptions()
+
+	// Set Resource Group ID
+	subnetOption.SetResourceGroupID(resourceGroupID)
+
+	// Get a list of all subnets
+	subnetList, _, err := c.vpcService.ListSubnets(subnetOption)
+	if err != nil {
+		return "", err
+	}
+
+	if subnetList != nil {
+		var subnetID string
+		for _, eachSubnet := range subnetList.Subnets {
+			if *eachSubnet.Name == subnetName {
+				// Get Subnet ID
+				subnetID = *eachSubnet.ID
+				return subnetID, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("Could not retrieve Subnet ID of name: %v", subnetName)
+}
+
+// GetSecurityGroupsByName retrieves Security Groups ID
+func (c *ibmCloudClient) GetSecurityGroupsByName(securityGroupNames []string, resourceGroupID string, vpcID string) ([]vpcv1.SecurityGroupIdentityIntf, error) {
+	// Initialize a map with Security Group Names
+	securityGroupMap := map[string]string{}
+	for _, item := range securityGroupNames {
+		securityGroupMap[item] = ""
 	}
 
-	// Update image when found a name match and its status is available
-	var image *vpcv1.Image
-	for _, eachImage := range privateImages.Images {
-		if *eachImage.Name == imageName && *eachImage.Status == vpcv1.ImageStatusAvailableConst {
-			image = &eachImage
-			return image, nil
+	// Initialize List Security Groups Options
+	securityGroupOptions := c.vpcService.NewListSecurityGroupsOptions()
+	// Set Resource Group ID
+	securityGroupOptions.SetResourceGroupID(resourceGroupID)
+	// Set VPC ID
+	securityGroupOptions.SetVPCID(vpcID)
+
+	// Get a List of Security Groups
+	securityGroups, _, _ := c.vpcService.ListSecurityGroups(securityGroupOptions)
+
+	var SecurityGroupIdentityList = make([]vpcv1.SecurityGroupIdentityIntf, len(securityGroupNames))
+	if securityGroups != nil {
+		idxCounter := 0
+		for _, eachSecurityGroup := range securityGroups.SecurityGroups {
+			if _, ok := securityGroupMap[*eachSecurityGroup.Name]; ok {
+				SecurityGroupIdentityList[idxCounter] = &vpcv1.SecurityGroupIdentityByID{
+					ID: eachSecurityGroup.ID,
+				}
+				idxCounter++
+			}
 		}
 	}
-	return nil, fmt.Errorf("Image: %s not found in Region: %s or Image may not be available yet", imageName, region)
+
+	// Check if retrived all IDs
+	if len(securityGroupNames) == len(SecurityGroupIdentityList) {
```

Comment:
Since the `SecurityGroupIdentityList` slice references the make array of size `len(securityGroupNames)`, won't these lengths be equal even if you only found 1 of 2 subnet IDs?
Also, `retrived` is misspelled in the above comment.

Reply:
Nice catch - Updating the Slice len() to 0